### PR TITLE
eric6: add upstream version 19.02

### PR DESCRIPTION
**Pull request by giansalvo**
###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS  10.13.4 17E202
Xcode 9.3.1 9E501

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by mf2k** [comment]
This should reference the Trac ticket https://trac.macports.org/ticket/56636 in the commit message.

**Comment by mf2k** [comment]
This port is fundamentally different from what was being discussed in that ticket. Namely:
 - We wanted use the python portgroup so that the desired python version can be used. This portfile only uses python 3.7. Perhaps that's not so bad though. Look at the py-spyder port as an example of this. 
 - This does away with the Qt portgroups/variants and just assumes the use of the old qt4. If we go this route, it is much better to use Qt5 and the Qt5 portgroup.

File: devel/eric6/Portfile [hunk 1]
```diff
@@ -0,0 +1,58 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:filetype=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem      1.0
+
+name            eric6
+version         19.02
+revision        1
```
Comment (mf2k) [inline devel/eric6/Portfile]: The revision starts at 0, not 1.

File: devel/eric6/Portfile [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+                size    16881300
+
+depends_lib     port:python37 \
+                port:py37-pyqt4 \
+                port:py37-qscintilla-qt4 \
+                port:py37-pysvn \
+                port:py37-rope \
+                port:py37-pylint \
+                port:py37-cx_Freeze \
+                port:py37-enchant
```
Comment (mf2k) [inline devel/eric6/Portfile]: Please sort the dependencies alphabetically.

File: devel/eric6/Portfile (hunk 1 above)
Comment (giansalvo) [reply devel/eric6/Portfile]: @mf2k 
Hi, I amended the Portfile as per your comments. I'm new to GitHub workflow, how can I submit the new file? Should I open a new PR or can I use this one?
Thanks

**Comment by mf2k** [comment]
Don't submit a new PR. You can commit the changes to your branch.

**Comment by giansalvo** [comment]
I pushed a new commit on my branch.

File: devel/eric6/Portfile [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    flexible Scintilla editor control. It is designed to be usable as everdays' \
+    quick and dirty editor as well as being usable as a professional project \
+    management tool integrating many advanced features Python offers the \
+    professional coder.
+
+homepage        http://eric-ide.python-projects.org/index.html
+master_sites    sourceforge:project/eric-ide/${name}/stable/${version}/
+
+checksums       rmd160  6289a33cc10a13c50c901589e7842b6f3d030bb1 \
+                sha1    019992c5b31db30621c574175497a0fdd0ac255b \
```
Comment (pmetzger) [inline devel/eric6/Portfile]: A sha1 checksum isn't really necessary; we discourage them at this point.

**Comment by giansalvo** [comment]
>     * This does away with the Qt portgroups/variants and just assumes the use of the old qt4. If we go this route, it is much better to use Qt5 and the Qt5 portgroup.
@mf2k 
I tried using Qt5 but ended up with an error (note that PyQt5 is installed and recognized):
:info:destroot Checking dependencies
:info:destroot Python Version: 3.7.2
:info:destroot Found PyQt5
[...]
:info:destroot Sorry, please install PyQt5.QtWebEngineWidgets or PyQt5.QtWebKitWidgets.
I can't find the right packages for PyQt5.QtWebEngineWidgets or PyQt5.QtWebKitWidgets. Please help.

**Comment by mf2k** [comment]
Looking at the qt5 portgroup you need to add something like the following:

```
PortGroup                  qt5 1.0
 
...

qt5.depends_component qtwebengine
qt5.depends_component qtwebkit
```

**Comment by iEFdev** [comment]
> This should reference the Trac ticket https://trac.macports.org/ticket/56636 in the commit message.

@mf2k : You can add: “Closes: https://trac.macports.org/ticket/52865” as well, or if you close it now. Never got started with it. _(some Qt/Lion thing)._

**Comment by reneeotten** [comment]
this PR seems to have stalled... anyone still working on getting this fixed? If not, I can take a look over the weekend or so.

**Comment by pmetzger** [comment]
@reneeotten I encourage you to do so. My time for handling these things has gotten far more limited.

**Pull request closed without merging**